moveit_experimental/CMakeLists.txt

```diff
@@ -56,11 +57,10 @@ catkin_package(
     pluginlib
   DEPENDS
     Boost
-    Eigen3
```

Comment:
`INCLUDE_DIRS` should only contain

> `CMAKE_CURRENT_SOURCE_DIR`-relative paths to C/C++ includes

So adding `EIGEN3_INCLUDE_DIR` is not permitted (as @v4hn also mentioned in https://github.com/ros-planning/moveit/pull/254#issuecomment-249788102). `DEPENDS` is used to export non-ROS pkg includes/libraries (from [catkin_package()](http://docs.ros.org/indigo/api/catkin/html/dev_guide/generated_cmake_api.html#catkin-package)).